component/prometheus/exporter/azure/azure.go

```diff
@@ -0,0 +1,76 @@
+package azure
+
+import (
+	"github.com/grafana/agent/component"
+	"github.com/grafana/agent/component/prometheus/exporter"
+	"github.com/grafana/agent/pkg/integrations"
+	"github.com/grafana/agent/pkg/integrations/azure_exporter"
+)
+
+func init() {
+	component.Register(component.Registration{
+		Name:          "prometheus.exporter.azure",
+		Args:          Arguments{},
+		Exports:       exporter.Exports{},
+		NeedsServices: exporter.RequiredServices(),
+		Build:         exporter.New(createExporter, "azure"),
+	})
+}
+
+func createExporter(opts component.Options, args component.Arguments) (integrations.Integration, error) {
+	a := args.(Arguments)
+	return a.Convert().NewIntegration(opts.Logger)
+}
+
+type Arguments struct {
+	Subscriptions            []string `river:"subscriptions,attr"`
+	ResourceGraphQueryFilter string   `river:"resource_graph_query_filter,attr,optional"`
+	ResourceType             string   `river:"resource_type,attr"`
+	Metrics                  []string `river:"metrics,attr"`
+	MetricAggregations       []string `river:"metric_aggregations,attr,optional"`
+	Timespan                 string   `river:"timespan,attr,optional"`
+	IncludedDimensions       []string `river:"included_dimensions,attr,optional"`
+	IncludedResourceTags     []string `river:"included_resource_tags,attr,optional"`
+	MetricNamespace          string   `river:"metric_namespace,attr,optional"`
+	MetricNameTemplate       string   `river:"metric_name_template,attr,optional"`
+	MetricHelpTemplate       string   `river:"metric_help_template,attr,optional"`
+	AzureCloudEnvironment    string   `river:"azure_cloud_environment,attr,optional"`
+}
+
+var DefaultArguments = Arguments{
+	Timespan:              "PT1M",
+	MetricNameTemplate:    "azure_{type}_{metric}_{aggregation}_{unit}",
+	MetricHelpTemplate:    "Azure metric {metric} for {type} with aggregation {aggregation} as {unit}",
```

Comment:
Not 100% sure, but do you think we should make this into `prometheus_exporter_azure`, or would it be too long?

Reply:
if we change this default value it could be a breaking change for the people migrating from static to flow, no?